lib/ansible/modules/cloud/amazon/lightsail.py

```diff
@@ -0,0 +1,405 @@
+    client: authenticated lightsail connection object
+    instance_name: name of instance to reboot
+
+    Returns a dictionary of instance information
+    about the restarted instance
+
+    If the instance was not able to reboot,
+    "changed" will be set to False.
+
+    Wait will not apply here as this is an OS-level operation
+    """
+    wait = module.params.get('wait')
+    wait_timeout = int(module.params.get('wait_timeout'))
+    wait_max = time.time() + wait_timeout
+
+    changed = False
+
+    inst = None
+    try:
+        inst = find_instance_info(client, instance_name)
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] != 'NotFoundException':
+            module.fail_json(msg='Error finding instance {0}, error: {1}'.format(instance_name, e))
+
+    # Wait for instance to exit transition state before state change
+    if wait:
+        while wait_max > time.time() and inst is not None and inst['state']['name'] in ('pending', 'stopping'):
+            try:
+                time.sleep(5)
+                inst = find_instance_info(client, instance_name)
+            except botocore.exceptions.ClientError as e:
+                time.sleep(3)
+
+    # send reboot
+    if inst is not None:
+        try:
+            client.reboot_instance(instanceName=instance_name)
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] != 'NotFoundException':
+                module.fail_json(msg='Unable to reboot instance {0}, error: {1}'.format(instance_name, e))
+        changed = True
+
+    return (changed, inst)
+
+def startstop_instance(module, client, instance_name, state):
+    """
+    Starts or stops an existing instance
+
+    module: Ansible module object
+    client: authenticated lightsail connection object
+    instance_name: name of instance to start/stop
+    state: Target state ("running" or "stopped")
+
+    Returns a dictionary of instance information
+    about the instance started/stopped
+
+    If the instance was not able to state change,
+    "changed" will be set to False.
+
+    """
+    wait = module.params.get('wait')
+    wait_timeout = int(module.params.get('wait_timeout'))
+    wait_max = time.time() + wait_timeout
+
+    changed = False
+
+    inst = None
+    try:
+        inst = find_instance_info(client, instance_name)
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] != 'NotFoundException':
+            module.fail_json(msg='Error finding instance {0}, error: {1}'.format(instance_name, e))
+
+    # Wait for instance to exit transition state before state change
+    if wait:
+        while wait_max > time.time() and inst is not None and inst['state']['name'] in ('pending', 'stopping'):
+            try:
+                time.sleep(5)
+                inst = find_instance_info(client, instance_name)
+            except botocore.exceptions.ClientError as e:
+                time.sleep(1)
```

Comment:
Is it possible to narrow the exceptions handled here? This loop could just keep retrying over and over for an error that won't be fixed by waiting longer.